Boost fails to build on RHEL 6

Here is the full log: https://gist.githubusercontent.com/certik/9fa79281781d07da5eff/raw/build.log

the issue is:

```
2014/08/07 17:18:40 - INFO: [package:run_job] ./boost/python/detail/wrap_python.hpp:50:23: error: pyconfig.h: No such file or directory
2014/08/07 17:18:40 - INFO: [package:run_job] ./boost/python/detail/wrap_python.hpp:75:24: error: patchlevel.h: No such file or directory
2014/08/07 17:18:40 - INFO: [package:run_job] ./boost/python/detail/wrap_python.hpp:78:2: error: #error Python 2.2 or higher is required for this version of Boost.Python.
2014/08/07 17:18:40 - INFO: [package:run_job] ./boost/python/detail/wrap_python.hpp:142:21: error: Python.h: No such file or directory
```

I have gcc 4.4.7. So I think it just can't find the Python headers.

--- a/pkgs/boost/boost.yaml
+++ b/pkgs/boost/boost.yaml
@@ -2,9 +2,9 @@ extends: [base_package]
 
 dependencies:
   when platform == 'linux':
-    build: [bzip2, mpi, zlib, patchelf]
+    build: [bzip2, mpi, zlib, patchelf, {{build_with}}]
   when platform != 'linux':
-    build: [bzip2, mpi, zlib]
+    build: [bzip2, mpi, zlib, {{build_with}}]
 
 sources:
 - url: http://downloads.sourceforge.net/boost/boost_1_55_0.tar.bz2

